include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,989 @@
+                    // set shared sums of x^2, ux^2, (x*ux)^2, particle counter to zero
+                    shSumMom2[linearIdx] = 0.0_X;
+                    shSumPos2[linearIdx] = 0.0_X;
+                    shSumMomPos2[linearIdx] = 0.0_X;
+                    shCount_e[linearIdx] = 0.0_X;
+                }
+            );
+            __syncthreads( );
+
+            DataSpace< simDim > const superCellIdx( mapper.getSuperCellIndex(
+                DataSpace< simDim >( blockIdx )
+            ));
+
+            // each virtual thread is working on an own frame
+            FramePtr frame = pb.getLastFrame( superCellIdx );
+
+            // end kernel if we have no frames within the supercell
+            if( !frame.isValid( ) )
+                return;
+
+            auto accFilter = filter(
+                acc,
+                superCellIdx - mapper.getGuardingSuperCells( ),
+                WorkerCfg< numWorkers >{ workerIdx }
+            );
+
+            memory::CtxArray<
+                typename FramePtr::type::ParticleType,
+                ParticleDomCfg
+            >
+            currentParticleCtx(
+                workerIdx,
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    auto particle = frame[ linearIdx ];
+                    /* - only particles from the last frame must be checked
+                     * - all other particles are always valid
+                     */
+                    if( particle[ multiMask_ ] != 1 )
+                        particle.setHandleInvalid( );
+                    return particle;
+                }
+            );
+
+            while( frame.isValid( ) )
+            {
+                // loop over all particles in the frame
+                ForEachIdx< ParticleDomCfg > forEachParticle( workerIdx );
+
+                forEachParticle(
+                    [&](
+                        uint32_t const linearIdx,
+                        uint32_t const idx
+                    )
+                    {
+                        /* get one particle */
+                        auto & particle = currentParticleCtx[ idx ];
+                        if(accFilter(acc,particle)){
+
+                            float_X const weighting = particle[ weighting_ ];
+                            float3_X const mom = particle[ momentum_ ] / weighting;
+                            floatD_X const pos = particle[ position_ ];
+
+                            lcellId_t const cellIdx = particle[ localCellIdx_ ];
+                            DataSpace<simDim> const frameCellOffset(DataSpaceOperations<simDim>::template map<MappingDesc::SuperCellSize > (cellIdx));
+                            auto const localSupercellStart = (superCellIdx - mapper.getGuardingSuperCells()) * MappingDesc::SuperCellSize::toRT();
+                            int const index_y = frameCellOffset.y() + localSupercellStart.y();
```

Comment:
BUG: This create an out of memory access in shared memory. A valid range for `index_y` is [0;SuperCellSize::y]` see https://github.com/ComputationalRadiationPhysics/picongpu/pull/2588/files#diff-f1fb3cee53e1f5187bf099a126e22114R115